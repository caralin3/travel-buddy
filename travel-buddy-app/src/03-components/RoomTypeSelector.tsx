import React, { ChangeEvent } from 'react';
import { Select, SelectOption } from '../02-molecules';

export interface RoomTypeSelectorProps {
  id: string;
  label: string;
  onSelect: (e: ChangeEvent<HTMLInputElement>) => void;
  required?: boolean;
  value: string;
}

export const RoomTypeSelector: React.FC<RoomTypeSelectorProps> = (props) => {
  const options: SelectOption[] = [
    {
      label: 'Single',
      value: 'SINGLE',
    },
    {
      label: 'Double',
      value: 'DOUBLE',
    },
    {
      label: 'Suite',
      value: 'SUITE',
    },
  ];

  return <Select options={options} {...props} />;
};
