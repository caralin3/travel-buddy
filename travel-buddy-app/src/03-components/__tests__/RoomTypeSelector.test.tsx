import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { RoomTypeSelector } from '../RoomTypeSelector';

describe('RoomTypeSelector', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<RoomTypeSelector id="room" label="Select" onSelect={jest.fn()} value="SINGLE" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
