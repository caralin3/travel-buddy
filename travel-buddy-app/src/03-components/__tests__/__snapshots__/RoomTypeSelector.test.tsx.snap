// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomTypeSelector should render correctly 1`] = `
<Select
  id="room"
  label="Select"
  onSelect={[MockFunction]}
  options={
    Array [
      Object {
        "label": "Single",
        "value": "SINGLE",
      },
      Object {
        "label": "Double",
        "value": "DOUBLE",
      },
      Object {
        "label": "Suite",
        "value": "SUITE",
      },
    ]
  }
  value="SINGLE"
/>
`;
